### ISSUE TITLE
Unable to add two lists of the same type

### ISSUE THREAD

**Issue by SandyChapman** [microsoft/pyright#3017]
I'm working with a 3rd party library that uses a sentinel object to detect default value usage on an Optional initializer parameter. However, when I attempt to concatenate two lists of these objects with the `+` operator, I'm getting a error reported by Pyright and Pylance.

Reproducible code:

```python

from typing import List

_DEFAULT_VALUE = object()

class MyClass(object):

    def __init__(
        self,
        name,
        description=_DEFAULT_VALUE,
    ):
        ...

X: List[MyClass] = [MyClass("a", description="b")]
Y: List[MyClass] = [MyClass("c")]
Z: List[MyClass] = X + Y # Operator "+" not supported for types "list[MyClass]" and "list[MyClass]"
```

The error is shown on the last line `Operator "+" not supported for types "list[MyClass]" and "list[MyClass]"`

The interesting part is if I remove the `description` param when assigning the list to `X` in the example above, the error goes away. 

Pyright 1.1.220 and Pylance v2022.2.0 on MacOS

![image](https://user-images.githubusercontent.com/3920249/153290853-ed8a16ed-aee9-4760-a223-1768548270e2.png)

**Comment by SandyChapman** [comment]
Ah, that's really interesting and makes sense as a desirable default behaviour. Thanks again for your quick turnaround!

**Comment by erictraut** [comment]
This is included in pyright 1.1.221, which I just published. It will also be included in the next release of pylance.
